Client/Frontend/Home/PanelDataObservers.swift

```diff
@@ -65,7 +65,8 @@ class ActivityStreamDataObserver: DataObserver {
         }
 
         // Highlights are cached for 15 mins
-        let shouldInvalidateHighlights = highlights || (Timestamp.uptimeInMilliseconds() - lastInvalidation > invalidationTime)
+        let userEnabledHighlights = profile.prefs.boolForKey(PrefsKeys.ASRecentHighlightsVisible) ?? true
+        let shouldInvalidateHighlights = (highlights || (Timestamp.uptimeInMilliseconds() - lastInvalidation > invalidationTime)) && userEnabledHighlights
```

Comment:
Wouldn't we still want to invalidate if they're turned off?

Reply:
invalidating highlights will fetch and cache the highlights. I wanted to avoid doing unnecessary sql.